model/modelindexer/indexer.go

```diff
@@ -309,13 +318,7 @@ func encodeMap(v map[string]interface{}, out *fastjson.Writer) error {
 	return nil
 }
 
-func (i *Indexer) flushActive() {
-	i.activeMu.Lock()
-	defer i.activeMu.Unlock()
-	i.flushActiveLocked(context.Background())
-}
-
-func (i *Indexer) flushActiveLocked(ctx context.Context) {
+func (i *Indexer) flushActive(ctx context.Context) error {
 	// Create a child context which is cancelled when the context passed to i.Close is cancelled.
 	flushed := make(chan struct{})
```

Comment:
the `flushed` channel is never closed, and it never receives anything, shouldn't we be sending a `struct{}{}` after the`i.Flush()` call (or `close(flushed)`) so the context is cancelled right away and the goroutine exits? Otherwise we may leak goroutines when we this method is called.

Reply:
Yep, I just forgot to move the `defer close(flushed)` out of the old goroutine. Thanks, good catch.

Reply:
Thanks, I've adapted your test a little bit and included it with the fix.